### ISSUE TITLE
Ability to toggle (enable/disable) LDAP configuration from within the admin UI?

### ISSUE THREAD

**Issue by Boilerplate4u** [owncloud/core#24762]
(Also mailed to devel@owncloud.org)

core version: 9.0.2
ldap version 0.8
- Subject:
  Is it deliberately that a "button" is missing in the LDAP admin panel to enable/disable a LDAP configuration?
- Background
  The activation of LDAP configuration seems to be fully automatic and there is no way to disable it. During LDAP configuration "view.js" will continuously check the status  with _updateStatusIndicator: function(state)  if (updateStatusIndicator == STATUS_SUCCESS) which sets ldap_configuration_active => 1 and activates LDAP user accounts and user mappings according to the LDAP Filters. **But there is no way to inactivate the configuration from within the admin UI other than delete it and start over from scratch**. You can of course "hack" the config.php with ldapConfigurationActive or update "ldap_configuration_active" in table oc_appconfig but that's beside the point.
- The issue and root cause
  The problem with the automation is that it activates the configuration to soon, especially when you also need to configure some additional settings in the expert panel like UUID mappings or the Internal Username Attribute that all affects the internal user naming conventions. On order to make these settings to work you need to reset the configuration i.e "flush" (reset) the UUID mapping as well as remove the wrongly injected LDAP user accounts in the database thus the need to be able to disable the automatic "check-and-activate" and add an [Enable/Disable] button.
- Bottom line
  MO there is no need to "over-automate" a function like LDAP that anyhow require expert knowledge to make it work.
- What's next?
  I need some advice regarding this matter: should I report this as a bug or just try to convince someone to add an [Enable/Disable] button in the LDAP panel as an additional "feature" ??
## 

Thanks in advance!
Regards, Lars.

**Comment by PVince81** [comment]
@blizzz @davicente 

I don't think there is a way to temporarily disable LDAP server configs in the UI, so this is mostly a new feature. Tagging as such.

CC @MTRichards

**Comment by Boilerplate4u** [comment]
Well, this "feature" would actually fix a major issue with most settings found in the LDAP Advanced and Expert panel. Currently you cannot add or alter any attributes that affects the internal user mappings because the current automatic activation "feature" always sets these values beforehand and hence hindering further changes.

So with the current functionality you can more or less disable the Advanced and Expert panel since they cannot be used as it is of today.

Although, one interesting fact that might be worth investigating: Under "Advanced" -> Connection Settings there is a "Configuration Active" check box but it seems more related to "Backup (Replica) Host" or is it???

**Comment by blizzz** [comment]
> Well, this "feature" would actually fix a major issue related to most settings found in the LDAP Advanced and Expert panel. Currently you cannot add or alter any attributes that affects the internal user mappings because the current automatic activation "feature" always sets these values beforehand and hence hindering further changes.

No issue like this is yet reported.

> Well, this "feature" would actually fix a major issue related to most settings found in the LDAP Advanced and Expert panel. Currently you cannot add or alter any attributes that affects the internal user mappings because the current automatic activation "feature" always sets these values beforehand and hence hindering further changes.

https://doc.owncloud.org/server/9.0/admin_manual/configuration_user/user_auth_ldap.html#connection-settings

**Comment by Boilerplate4u** [comment]
> No issue like this is yet reported.

If you look around, you'll probably find related issues that forced them to use [Clear Username-LDAP User Mapping] or [Clear Groupname-LDAP Group Mapping] as a result of this...

_"Enables or Disables the current configuration. By default, it is turned off. When ownCloud makes a successful test connection it is automatically turned on"_

There you go, thank you! IMO, but a rather misplaced location if I may say :-)

Suggestions:

1. Move the check-box "Configuration Active[x]" to the regular standard panel.
2. Change or find out a way of working to allow changes of the advanced mapping attributes before the user mapping tables are populated.
--

Thanks, Lars.

**Comment by jvillafanez** [comment]
Related to https://github.com/owncloud/core/issues/24762

**Comment by PVince81** [comment]
Please reopen in [user_ldap app repo](https://github.com/owncloud/user_ldap) if issue persists with 10.0.4 and the latest user_ldap app from marketplace.
